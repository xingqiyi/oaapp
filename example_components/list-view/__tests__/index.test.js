import React from 'react';
import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import ListView from '../index';

describe('ListView', () => {
  it('renders correctly', () => {
    class Minimal extends React.Component {
      constructor() {
        super();
        const ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 !== r2 });
        this.state = {
          dataSource: ds.cloneWithRows(['row 1', 'row 2']),
        };
      }
      render() {
        return (
          <ListView
            dataSource={this.state.dataSource}
            renderRow={rowData => <span>{rowData}</span>}
          />
        );
      }
    }
    const wrapper = render(<Minimal />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });
});
