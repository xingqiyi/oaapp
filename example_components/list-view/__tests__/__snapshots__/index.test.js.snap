// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView renders correctly 1`] = `
<div
  class="am-list-view-scrollview"
  style="position:relative;overflow:auto;-webkit-overflow-scrolling:touch;flex:1;"
>
  <div
    class="am-list-view-scrollview-content am-list"
    style="position:absolute;min-width:100%;"
  >
    <div
      class="am-list-body"
    >
      <div
        class="list-view-section-body"
      >
        <span>
          row 1
        </span>
        <span>
          row 2
        </span>
      </div>
    </div>
  </div>
</div>
`;
